api/inc/page_allocator_exports.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2016, ARM Limited, All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef __UVISOR_API_PAGE_ALLOCATOR_EXPORTS_H__
+#define __UVISOR_API_PAGE_ALLOCATOR_EXPORTS_H__
+
+#include <stdint.h>
+
+/* Must be a power of 2 for MPU alignment in ARMv7-M */
+#ifndef UVISOR_PAGE_SIZE
```

Comment:
Since this is an `_exports` file, it is used by both uVisor internally and by the host OS. The definitions should not be conditional, otherwise the uVisor core (which is pre-compiled) will work with wrong assumptions.

There are 3 options here:
1. Make the value fixed, forever.
1. Make the value uVisor-target-dependent. It then becomes part of the uVisor porting process to decide the default (and fixed) value.
1. It is app-dependent. In this case the value must be read at runtime by uVisor from flash. It must be part of the config table that uVisor uses to gather info from the host.

This applies to all `ifdef`s here.